plugins/woocommerce-admin/client/products/fields/attribute-term-input-field/create-attribute-term-modal.tsx

```diff
@@ -0,0 +1,162 @@
+import { cleanForSlug } from '@wordpress/url';
+import { Form, FormContext, FormErrors } from '@woocommerce/components';
+import { recordEvent } from '@woocommerce/tracks';
+import {
+	EXPERIMENTAL_PRODUCT_ATTRIBUTE_TERMS_STORE_NAME,
+	ProductAttributeTerm,
+	QueryProductAttribute,
+} from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import './create-attribute-term-modal.scss';
+
+type CreateAttributeTermModalProps = {
+	initialAttributeTermName: string;
+	attributeId: number;
+	onCancel: () => void;
+	onCreated: ( newAttribute: ProductAttributeTerm ) => void;
+};
+
+export const CreateAttributeTermModal: React.FC<
+	CreateAttributeTermModalProps
+> = ( { initialAttributeTermName, attributeId, onCancel, onCreated } ) => {
+	const { createNotice } = useDispatch( 'core/notices' );
+	const [ isCreating, setIsCreating ] = useState( false );
+	const { createProductAttributeTerm, invalidateResolutionForStoreSelector } =
+		useDispatch( EXPERIMENTAL_PRODUCT_ATTRIBUTE_TERMS_STORE_NAME );
+
+	const onAdd = async ( attribute: Partial< ProductAttributeTerm > ) => {
+		recordEvent( 'product_attribute_term_add', {
+			new_product_page: true,
+		} );
+		setIsCreating( true );
+		try {
+			const newAttribute: ProductAttributeTerm =
+				await createProductAttributeTerm( {
+					...attribute,
+					attribute_id: attributeId,
+				} );
+			recordEvent( 'product_attribute_term_add_success', {
+				new_product_page: true,
+			} );
+			invalidateResolutionForStoreSelector( 'getProductAttributes' );
+			setIsCreating( false );
+			onCreated( newAttribute );
+		} catch ( e ) {
+			recordEvent( 'product_attribute_term_add_failed', {
+				new_product_page: true,
+			} );
+			createNotice(
+				'error',
+				__( 'Failed to create attribute term.', 'woocommerce' )
+			);
+			setIsCreating( false );
+			onCancel();
+		}
+	};
+
+	function validateForm(
+		values: Partial< ProductAttributeTerm >
+	): FormErrors< ProductAttributeTerm > {
+		const errors: FormErrors< ProductAttributeTerm > = {};
+
+		if ( ! values.name?.length ) {
+			errors.name = __(
+				'The attribute term name is required.',
+				'woocommerce'
+			);
+		}
+
+		return errors;
+	}
+
+	return (
+		<Modal
+			title={ __( 'Create attribute', 'woocommerce' ) }
+			onRequestClose={ () => onCancel() }
+			className="woocommerce-create-attribute-term-modal"
+		>
+			<Form< Partial< ProductAttributeTerm > >
+				initialValues={ {
+					name: initialAttributeTermName,
+					slug: cleanForSlug( initialAttributeTermName ),
+				} }
+				validate={ validateForm }
+				errors={ {} }
+				onSubmit={ onAdd }
+			>
+				{ ( {
+					getInputProps,
+					handleSubmit,
+					isValidForm,
+					setValue,
+					values,
+				}: FormContext< QueryProductAttribute > ) => {
+					const nameInputProps = getInputProps< string >( 'name' );
+					return (
+						<>
+							<TextControl
+								label={ __( 'Name', 'woocommerce' ) }
+								{ ...nameInputProps }
+								onBlur={ () => {
```

Comment:
If the `onBlur` here fires, it actually closes both the terms modal and the add attribute modal, which I'd guess wouldn't be expected. Should we stop it from bubbling down to the add modal as well?

Reply:
Hmm could you maybe share a video of this? Things work as expected for me and the Modal doesn't auto close for me when the `onBlur` is being fired.